### PR TITLE
[CAD-691] GitHub actions migration

### PR DESCRIPTION
**Pull request by jbropho**
* env vars added for kafka hostname etc in tests; could be accessed via localhost in circleci, cannot in github actions
* license added to project.clj, lein auto adds this to the pom when present, became mandatory to have a license in your pom when pushing to clojars on 23/12/23 - https://clojureverse.org/t/ann-clojars-now-requires-a-license-in-the-pom-for-new-projects-or-projects-that-already-specify-a-license/10347/2?u=tcrawley
* github actions setup + circleci removed
* deploy step uses a protected env `clojars_deploy` - running this step requires approval from a required approver (@FundingCircle/team-credit-application and @FundingCircle/jackdaw) currently

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,6 +46,10 @@ Anyone with the appropriate credentials can "cut a release" of jackdaw using the
 
 Steps 2 to 6 is essentially `git tag $version -m "$title\n\n$description" && git push --tags`
 
+#### Snapshot release
+
+Snapshot releases can be created by pushing a tag with the format `publish-snapshot-semver`
```
Comment (james-doolan) [inline README.md:51]: Does this mean anyone can publish a snapshot release with any version they want?

File: project.clj [hunk 2]
```diff
@@ -33,12 +35,13 @@
...
   :aot [jackdaw.serdes.edn2 jackdaw.serdes.fressian jackdaw.serdes.fn-impl]
-  :plugins [[me.arrdem/lein-git-version "2.0.8"]]
+  :plugins [[me.arrdem/lein-git-version "2.0.8"]
+            [com.github.clj-kondo/lein-clj-kondo "0.2.5"]]
 
   :git-version
   {:status-to-version
    (fn [{:keys [tag branch ahead? dirty?] :as git}]
-     (if (and tag (not ahead?) (not dirty?))
+     (if (and tag (not ahead?) (not dirty?) (= "master" branch))
```
Comment (james-doolan) [inline project.clj:44]: Should this be handled with trigger filters in the workflow def?

File: test/jackdaw/test/commands/write_test.clj [hunk 3]
```diff
@@ -40,7 +40,7 @@
                    :key-serde :long
                    :value-serde :json}))
 
-(def kafka-config {"bootstrap.servers" "localhost:9092"
+(def kafka-config {"bootstrap.servers" "kafka:9092"
```
Comment (james-doolan) [inline test/jackdaw/test/commands/write_test.clj:43]:
```suggestion
(def kafka-config {"bootstrap.servers" (str (utils/bootstrap-servers) ":9092")
```

File: test/jackdaw/test/transports/rest_proxy_test.clj [hunk 4]
```diff
@@ -30,7 +31,7 @@
 (defn kstream-config
   [app app-id]
   {:topology app
-   :config {"bootstrap.servers" "localhost:9092"
+   :config {"bootstrap.servers" "kafka:9092"
             "application.id" app-id}})
```
Comment (james-doolan) [inline test/jackdaw/test/transports/rest_proxy_test.clj:35]:
```suggestion
   :config (merge kafka-config {"application.id" app-id})})
```

File: test/jackdaw/test_test.clj [hunk 5]
```diff
@@ -234,7 +235,7 @@
     (try
       (jd.test/with-test-machine (trns/transport {:type :mock
                                                   :driver (jd.test/mock-test-driver (echo-stream test-in test-out)
-                                                                                    {"bootstrap.servers" "localhost:9092"
+                                                                                    {"bootstrap.servers" (str (utils/bootstrap-servers) ":9092")
                                                                                      "application.id" "test-echo-stream"})
```
Comment (james-doolan) [inline test/jackdaw/test_test.clj:239]:
```suggestion
(assoc kafka-config "application.id" "test-echo-stream")
```
update other cases too

**Review by james-doolan — COMMENTED** [review]
Minor comments regarding consistent use of env vars

File: project.clj (hunk 2 above)
Comment (jbropho) [reply project.clj:44]: Added to make more explicit that all non-master builds are marked snapshot; it should be handled already by the current setup, happy to remove it it adds confusion.

File: README.md (hunk 1 above)
Comment (jbropho) [reply README.md:51]: Described slightly higher in the README, essentially:
* outside contributors cannot tag, can only open PRs from forks
* tags are protected
* deploy env is protected, deploy step cannot run without approval from jackdaw or credit-applications team review

**Review by james-doolan — APPROVED** [review]

**Pull request merged**